src/PeliasMapkitExtensions.swift

```diff
@@ -124,21 +124,23 @@ public extension PeliasResponse {
     //TODO: This should get refactored into eventually being a real GeoJSON decoder, and split out the MapItem creation
     var mapItems = [PeliasMapkitAnnotation]()
     if let jsonDictionary = parsedResponse?.parsedResponse {
-      let featuresArray = jsonDictionary["features"] as! [[String:AnyObject]]
+      guard let featuresArray = jsonDictionary.object(forKey: "features") as? [[String:AnyObject]] else {
+        return nil
+      }
       for feature in featuresArray {
         //Address Dictionary for Placemark Creation
-        let featureProperties = feature["properties"] as! [String:AnyObject]
+        let featureProperties = feature["properties"] as? [String:AnyObject]
         var addressDictionary = [String:String]()
-        addressDictionary[PeliasIDKey] = featureProperties["id"] as? String
-        addressDictionary[PeliasDataSourceKey] = featureProperties["source"] as? String
+        addressDictionary[PeliasIDKey] = featureProperties?["id"] as? String
```

Comment:
If by work you mean doesn't crash, then yes :) The way it works is that the key `PeliasIDKey` is not added to `addressDictionary` if there is not an "id" in the `featureProperties`. I think this is expected...but looking at `PeliasMapKitAnnotation`, it looks like we cast it to `[String:AnyObject]` so we can probably update addressDictionary and remove the string casting...will do that now